staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery_controller.go

```diff
@@ -180,6 +218,12 @@ func (c *DiscoveryController) sync(version schema.GroupVersion) error {
 
 	if !foundVersion {
 		c.versionHandler.unsetDiscovery(version)
+		if c.openAPIAggregationManager != nil {
+			// Remove the local API service for the CRD in apiExtensionsServer's
+			// openAPIAggregationManager. If the API service doesn't exist (e.g. because
+			// CRD Validation OpenAPIV3Schema is nil), no error will be returned.
```

Comment:
is the intension is to not returning error or you just saying the RemoveAPISerivceSpec does not return error. I don't think it is in the contract of that function to never return error, it is just current implementation. If we need not to return error here, we should do so by ignoring returned error from that function otherwise revise the comment please.